pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go

```diff
@@ -417,13 +488,35 @@ func (r *ReconcileQuarksSecret) createSecret(ctx context.Context, qsec *qsv1a1.Q
 
 	secret.SetLabels(secretLabels)
 
+	// If this is a copy (lives in a different namespace), we only do an update,
```

Comment:
I'd prefer it if this block could live somewhere else, because the `createSecret` func is called from several places like a generic helper, but now has some added functionality to do something else entirely.

The if block doesn't seem to need the labels from line 489, maybe `createSecrets` could  use a  specialized `createSecretCopies` instead?